index.js

```diff
@@ -66,6 +70,18 @@ List.prototype.getAnswer = function() {
 };
 
 /**
+ * overriding "when" function to avoid setting a value when there
+ * is not a default value and the question was not asked
+ */
+List.prototype.when = function() {
+  if (this._when && !this._when.apply(this, arguments)) {
```

Comment:
It's possible for `this.when` to return a promise, so the results need to be resolved first:

```js
return Promise.resolve(Radio.prototype.when.apply(this, arguments))
  .then(function(when) {
    ...
    return when;
  });
```